src/coreComponents/mainInterface/GeosxState.cpp

```diff
@@ -74,12 +74,10 @@ class Timer
 //////////////////////////////////////////////////////////////////////////////////////////////////////////////////
 string durationToString( std::chrono::system_clock::duration const duration )
 {
+  // If we want to print HH::MM::SS (maybe in addition to seconds-only):
+  // return GEOSX_FMT( "{:%T}", duration );
   double const seconds = std::chrono::duration_cast< std::chrono::milliseconds >( duration ).count() / 1000.0;
-
-  char buffer[ 32 ];
-  std::snprintf( buffer, 32, "%20.3fs", seconds );
-
-  return buffer;
+  return GEOSX_FMT( "{:>20.3f}s", seconds );
```

Comment:
This looks better indeed.
But moreover removes the issue of the fixed sized buffer that may eventually burst someday.

Reply:
Indeed. My first goal is to remove all mentions of `sprintf` from code. Not only could they overflow and overwrite the stack (e.g. if the user chose a particularly long name for their solver/region/whatever), but even those places that (correctly) used `snprintf` would print truncated messages for no good reason. For performance-sensitive code that can't afford allocation of a string (like file output in `HypreMatrix`) I left the ability to reuse a local buffer via `GEOSX_FMT_TO` with a mandatory size argument.